lib/ansible/module_utils/network/iosxr/utils/utils.py

```diff
@@ -61,3 +62,188 @@ def dict_delete(base, comparable):
 
 def pad_commands(commands, interface):
     commands.insert(0, 'interface {0}'.format(interface))
+
+
+def remove_command_from_config_list(interface, cmd, commands):
+    # To delete the passed config
+    if interface not in commands:
+        commands.insert(0, interface)
+    commands.append('no %s' % cmd)
+    return commands
+
+
+def add_command_to_config_list(interface, cmd, commands):
+    # To set the passed config
+    if interface not in commands:
+        commands.insert(0, interface)
+    commands.append(cmd)
+
+
+def dict_diff(sample_dict):
+    # Generate a set with passed dictionary for comparison
+    test_dict = {}
+    for k, v in iteritems(sample_dict):
+        if v is not None:
+            if isinstance(v, list):
+                if isinstance(v[0], dict):
+                    li = []
+                    for each in v:
+                        for key, value in iteritems(each):
+                            if isinstance(value, list):
+                                each[key] = tuple(value)
+                        li.append(tuple(each.items()))
+                    v = tuple(li)
+                else:
+                    v = tuple(v)
+            elif isinstance(v, dict):
+                li = []
+                for key, value in iteritems(v):
+                    if isinstance(value, list):
+                        v[key] = tuple(value)
+                li.extend(tuple(v.items()))
+                v = tuple(li)
+            test_dict.update({k: v})
+    return_set = set(tuple(test_dict.items()))
+    return return_set
+
+
+def filter_dict_having_none_value(want, have):
+    # Generate dict with have dict value which is None in want dict
+    test_dict = dict()
+    test_key_dict = dict()
+    test_dict['name'] = want.get('name')
+    for k, v in iteritems(want):
+        if isinstance(v, dict):
+            for key, value in iteritems(v):
+                if value is None:
+                    dict_val = have.get(k).get(key)
+                    test_key_dict.update({key: dict_val})
+                test_dict.update({k: test_key_dict})
+        if v is None:
+            val = have.get(k)
+            test_dict.update({k: val})
+    return test_dict
+
+
+def remove_duplicate_interface(commands):
+    # Remove duplicate interface from commands
+    set_cmd = []
+    for each in commands:
+        if 'interface' in each:
+            if each not in set_cmd:
+                set_cmd.append(each)
+        else:
+            set_cmd.append(each)
+
+    return set_cmd
+
+
+def validate_ipv4(value, module):
```

Comment:
Can `ipaddress` be reused here? Please refer https://github.com/ansible/ansible/pull/59929/files#diff-0573c4d180b8184c07ae778232ecb216R21 and https://github.com/ansible/ansible/pull/59929/files#diff-0573c4d180b8184c07ae778232ecb216R120.

Reply:
It can be used but I think it's better to coninue with respective implementation coz in case of incorrect IP output wrt to `ipaddress` module isn't formatted at all which is not very user-friendly I feel, check the ref below:
output with `ipaddress` module: http://paste.openstack.org/show/762530/
output with existing fn: http://paste.openstack.org/show/762531/